sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2339,7 +2339,8 @@ object SQLConf {
       .doc("When false, the `strfmt` in `format_string(strfmt, obj, ...)` and " +
         "`printf(strfmt, obj, ...)` will no longer support to use \"0$\" to specify the first " +
         "argument, the first argument should always reference by \"1$\" when use argument index " +
-        "to indicating the position of the argument in the argument list.")
+        "to indicating the position of the argument in the argument list. " +
+        "This config will be removed in the future releases.")
```

Comment:
Since this doesn't work already, shall we put it in `removedSQLConfigs`?

Reply:
I think we should not do that because put the config to `removedSQLConfigs` means we shall remove it from `SQLConf` which leads to breaking existing user's code in two cases:
- User's code set the config to the default value `false` explicitly. In that case, user's app won't compile.
- User's code set it to `true` but don't refer to the zero index at all for some reasons - so user's might not face to `java.util.IllegalFormatArgumentIndexException`. But the removing the config, we break the app.

I believe we should deprecate the config first of all, and only after that we can remove it safely.

Reply:
+1 for the deprecation proposal. Maybe, could you add a proper warning to the SQL migration guide, too?

Reply:
I have updated the SQL migration guide.